src/server/connection.rs

```diff
@@ -1491,8 +1493,51 @@ impl Connection {
         self.video_ack_required = lr.video_ack_required;
     }
 
+    #[cfg(target_os = "windows")]
+    async fn handle_multiple_user_sessions(&mut self, lr: LoginRequest) {
+        if self.port_forward_socket.is_some() {
+            return;
+        } else {
+            let active_sessions = crate::platform::get_all_active_sessions();
+            if active_sessions.len() <= 1 {
+                return;
+            }
+            let usid;
+            match lr.option.user_session.parse::<u32>() {
+                Ok(n) => usid = Some(n),
+                Err(..) => usid = None,
+            }
+            let current_process_usid = crate::platform::get_current_process_session_id();
+            if usid.is_none() {
+                let mut res = LoginResponse::new();
+                let mut rdp = Vec::new();
+                for session in active_sessions {
+                    let u_sid = &session[0];
+                    let u_name = &session[1];
+                    let mut rdp_session = RdpUserSession::new();
+                    rdp_session.user_session_id = u_sid.clone();
+                    rdp_session.user_name = u_name.clone();
+                    rdp.push(rdp_session);
+                }
+                res.rdp_user_sessions = rdp;
+                let mut msg_out = Message::new();
+                msg_out.set_login_response(res);
+                self.send(msg_out).await;
+            }
+            if usid != self.user_session_id && usid != Some(current_process_usid) {
+                self.on_close("Reconnecting...", false).await;
+                std::thread::spawn(move || {
+                    let _ = ipc::connect_to_user_session(usid);
+                });
```

Comment:
Works fine, just a suggestion, `on_close` always exits the connection, it seems more reliable for `handle_multiple_user_sessions` to return false here and exit the connection without relaying on `--server` restart.